server/src/test/java/io/crate/integrationtests/CopyIntegrationTest.java

```diff
@@ -361,7 +364,17 @@ public void testCopyFromWithInvalidGivenGeneratedColumn() throws Exception {
                 " id int," +
                 " quote as cast(id as string)" +
                 ")");
-        execute("copy quotes from ? with (shared=true)", new Object[]{copyFilePath + "test_copy_from.json"});
+        // same with test_copy_from.json but with last entry with explicitly provided null
+        // in order to test that we can differentiate between "provided null" and "no value"
+        Path path = tmpFileWithLines(Arrays.asList(
+            """
+                {"id": 1, "quote": "Don't pañic."}
+                {"id": 2, "quote": "Would it save you a lot of time if I just gave up and went mad now?"}
+                {"id": 3, "quote": "Time is an illusion. Lunchtime doubly so."}
+                {"id": 4, "quote": null}
+            """
```

Comment:
Hi @seut, thank for catching the bug with `COALESCE` usage! I checked this test with provided null against prev version of the PR and it failed indeed. 

I introduced marker value in https://github.com/crate/crate/pull/14174/commits/5ea3b779c9f033e8ca9348a4ef6b50e1fd8067e7 to address this issue and also avoid `COALESCE` usage, could you please take another look?

PS. We also discussed another possible approach without marker:
>only collect known non-generated columns if the user does not provide target columns. if the user declare it, always collect value without any validation.

For JSON we don't have target columns so it would fall to the same behaviour as CSV without header and set `targetRefs` = all known columns (this is 5.2/5.3 behaviour) so for JSON we would always take the value without validation which would be a breaking change.

In https://github.com/crate/crate/pull/14221 we discussed that we might skip validation but AFAIK it's only about non-deterministic functions because of the use-case of `last update` timestamp. For deterministic I think we still want to keep 5.2/5.3 behaviour described in https://github.com/crate/crate/issues/14218#issue-1731935780? 
COPY FROM now takes the same code path as insert from subquery which does the validation for provided values.